source/slang/core.meta.slang

```diff
@@ -63,25 +63,24 @@ syntax unorm : UNormModifier;
 ///
 syntax snorm : SNormModifier;
 
-/// Modifier to indicate that a function name should not be mangled
+/// Modifier to indicate that a declaration's name should not be mangled
 /// by the Slang compiler.
 ///
-/// The `__unmangled` modifier should only be valid on functions
-/// and is mainly useful for the experimental heterogeneous
-/// features of Slang.
-///
 syntax __unmangled : UnmangledModifier;
 
-/// Modifier to indicate that a function name should be exported
-/// directly. Used in tandem with `__unmangled` in heterogeneous
-/// features of Slang.
+/// Modifier to indicate that a declaration's name should be exported
+/// directly. When being emitted, the mangled name is used directly
+/// and any Slang specific suffixes and prefixes are avoided.
 ///
-syntax __exportDirectly : __exportDirectly;
-
-/// Modifier to indicate that a struct is defined externally and
-/// should therefore not be exported by Slang.
+/// TODO: This currently contains a bad hack for passing through
+///       namespaces that relies on name hints. Potentially can be
+///       relegated to another existing decoration (intrinsic)
+/// TODO: The behavior that this modifier implies should technically
+///       be the default, whereas the current compiler behavior of
+///       generating unique names for most things should be
+///       considered a bit of a bug.
 ///
-syntax __externLib : __externLib;
+syntax __exportDirectly : ExportDirectlyModifier;
```

Comment:
Why do we still need this? It seems that after switching to __target_intrinsic this modifier is no longer used. We should consider remove it if that is the case.

Reply:
__exportDirectly is used in combination with __unmangled on the rest of the functions defined in shader.slang so that they are emitted unmangled. (Unless __target_intrinsic can also solve that problem?)

Reply:
I can swap to __target_intrinsic and remove the unneeded modifiers later today then (also remove the files you mentioned.)

Any idea what's causing the exception?

Reply:
We have made significant changes to the gfx library, so it is likely that this example is using gfx in an unsupported way that triggers D3D error.

Reply:
I can look into that as well then.

Is there some interaction between gfx and __target_intrinsic though? Because the example was running fine prior to switching from the __externLib modifier (I had fixed some gfx issues before this which I assume were from those gfx modification, but I very well could have missed a more recent change.)

Reply:
There shouldn't be any. I think it is just a gfx issue. If the crash is seen after you rebased / merged with master, you are also using a very different gfx library.

Reply:
Swapping __exportDirectly with __unmangled to __target_intrinsic is not viable because we want the function declaration to be emitted in the resulting shader.cpp (whereas we did not want the structs to be emitted.) Doing the swap led to identifier not found errors for the functions used in executeComputation (and when __target_intrinsic was applied to executeComputation then it was not emitted altogether.)

I can also confirm that my example works when I remove __target_intrinsic but keep the same previously generated shader.cpp, so the issue definitely is definitely caused (somehow) by the existence of __target_intrinsic. My gfx library usage matches all the other examples that were updated when the library changed, I updated to the new syntax before trying to swap to __target_intrinsic.